stacktrace.go

```diff
@@ -113,19 +113,40 @@ func (pcs *programCounters) Release() {
 	_stacktracePool.Put(pcs)
 }
 
-func (pcs *programCounters) Callers(skip int) []uintptr {
+// Callers captures program counters of callers of the current function.
+//
+// skip specifies the number of functions to skip.
+// limit specifies how deep into the stack we go, zero meaning unlimited.
+func (pcs *programCounters) Callers(skip int, limit int) []uintptr {
+	pc := pcs.pcs
+	if limit > 0 {
+		// The shortened slice will never be placed back into the
+		// pool, but the longer one will.
+		if limit < len(pc) {
+			pc = pc[:limit]
+		} else {
+			pc = make([]uintptr, limit)
```

Comment:
this case never gets hit in prod right? anyway we can refactor the change to avoid adding a case that's not necessary in prod?